### PR TITLE
Make Checkpoint classes data classes

### PR DESCRIPTION
**Pull request by roastario**

### CODE REVIEW

File: node/src/main/kotlin/net/corda/node/services/persistence/DBCheckpointStorage.kt [hunk 1]
```diff
@@ -127,7 +127,7 @@ class DBCheckpointStorage(
 
     @Entity
     @javax.persistence.Table(name = "${NODE_DATABASE_PREFIX}checkpoint_blobs")
-    class DBFlowCheckpointBlob(
+    data class DBFlowCheckpointBlob(
```
Comment (mnesbit) [inline node/src/main/kotlin/net/corda/node/services/persistence/DBCheckpointStorage.kt:130]: Why? If you are overriding hash and equals this adds nothing also limits evolution

Comment (roastario) [reply node/src/main/kotlin/net/corda/node/services/persistence/DBCheckpointStorage.kt:130]: you must override hashcode and equals when you have Array objects in your data class

Comment (roastario) [reply node/src/main/kotlin/net/corda/node/services/persistence/DBCheckpointStorage.kt:130]: The reason it should be data is, they are data objects - they have no functionality. Also, it's very useful to for the copy functionality as we need to take these objects and "blank" out certain fields before transforming them into RPC objects.

**Review by lankydan — APPROVED** [review]

File: node/src/main/kotlin/net/corda/node/services/persistence/DBCheckpointStorage.kt (hunk 1 above)
Comment (mnesbit) [reply node/src/main/kotlin/net/corda/node/services/persistence/DBCheckpointStorage.kt:130]: It also seals the classes which I why I hate data classes for anything that may evolve. Plus I write my own copy method often and it is trivial in kotlin:
fun copy(flowId:String = this.flowId, etc): DBFlowCheckpoint = DBFlowCheckpoint(flowId, etc)

**Pull request merged**